rasterio/warp.py

```diff
@@ -259,7 +259,10 @@ def reproject(source, destination=None, src_transform=None, gcps=None, rpcs=None
         raster to a destination of the same size is approximately
         56 MB. The default (0) means 64 MB with GDAL 2.2.
     kwargs:  dict, optional
-        Additional arguments passed to transformation function.
+        Additional arguments passed to both the image to image
+        transformer :cpp:func:`GDALCreateGenImgProjTransformer2` (for example,
+        MAX_GCP_ORDER=2) and the :cpp:struct:`GDALWarpOptions` (for example,
+        INIT_DEST=NO_DATA).
```

Comment:
How does this work without referencing the "gdal" item in the mapping? I don't know anything about intersphinx.

Reply:
https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html

> This extension can generate links to the documentation of objects in external projects, either explicitly through the external role, or as a fallback resolution for any other cross-reference.

It is currently using the "fallback resolution". It could be 
 made more explicit by using [external](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#role-external).